src/Computation/LazySolver.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Numerics;
+using Microsoft.Z3;
+using Symbolica.Expression;
+
+namespace Symbolica.Computation;
+
+internal sealed class LazySolver : ISolver
+{
+    private readonly IList<Action<ISolver>> _actions;
+    private ISolver? _solver;
```

Comment:
Can we not use `Lazy` here? Or are se worried about serialisation? Although I didn't think these were ever serialised. 

I guess right now we're also threadsafe because of the fact the `PooledSolver` is and that we only use a `Solver` on a particular branch / state before disposing of it anyway.

Reply:
Perhaps this will be clearer once the above misunderstanding is resolved. To be clear thought, `PooledSolver` is definitely not meant to be thread-safe (except for creating it using the pool of `Context`s).